packages/api/src/platforms/vtex/resolvers/offer.ts

```diff
@@ -57,10 +57,10 @@ export const StoreOffer: Record<string, Resolver<Root>> = {
 
     return null
   },
-  seller: (root) => {
+  seller: (root, _, ctx) => {
     if (isSearchItem(root)) {
       return {
-        identifier: root.seller.sellerId ?? '',
+        identifier: ctx.storage.channel?.seller || root.seller.sellerId || '',
```

Comment:
This doesn't smell good to me.

Suppose the offer isn't a offer from the seller in the channel. Setting this identifier here would mean this offer comes from the seller in the channel, which isn't true. 

Now suppose the offer is coming from the seller in the channel. Then this optional chain would be unnecessary, given that `root.seller.sellerId` would be equal to the channel's seller.

Reply:
> Suppose the offer isn't a offer from the seller in the channel. Setting this identifier here would mean this offer comes from the seller in the channel, which isn't true

This case is impossible because if `ctx.storage.channel?.seller` is non nullable, this means we fetched the offer from this specific seller, so both sellers match

> Now suppose the offer is coming from the seller in the channel. Then this optional chain would be unnecessary, given that root.seller.sellerId would be equal to the channel's seller.

Not really. Because of how the search API is constructed, `root.seller.sellerId` is the seller whitelabel, not really the real seller